src/main/java/com/jnape/palatable/shoki/api/MultiSet.java

```diff
@@ -107,4 +112,137 @@
+     * and <code>ys</code> with the occurrences of whichever one had less elements. This is equivalent to
+     * {@link MultiSet#merge} partially applied with {@link com.jnape.palatable.lambda.semigroup.builtin.Min#min}.
+     *
+     * @param other the {@link MultiSet} to intersect with this {@link MultiSet}
+     * @return the union {@link MultiSet}
+     * @see MultiSet#merge
+     */
+    default MultiSet<A> intersection(MultiSet<A> other) {
+        return merge(min(), other);
+    }
+
+
+    /**
+     * The symmetric difference of two {@link MultiSet MultiSets} <code>xs</code> and <code>ys</code> is the
+     * {@link MultiSet} of the occurrence elements present in either <code>xs</code> or <code>ys</code> but not the
+     * other.
+     *
+     * @param other the {@link MultiSet} to symmetrically subtract from this {@link MultiSet}
+     * @return the union {@link MultiSet}
+     * @see MultiSet#merge
+     */
+    default MultiSet<A> symmetricDifference(MultiSet<A> other) {
+        return difference(other).union(other.difference(this));
+    }
+
+    /**
+     * {@link Set} containing the unique elements from the {@link MultiSet}
+     *
+     * @return the unique {@link Set}
+     */
+    Set<Natural, A> unique();
+
+    /**
+     * Combine {@link MultiSet}s with a {@link Semigroup} to handle the combining of the amounts.
+     * Note that if one of the {@link MultiSet}s does not contain an {@link A} it will still pass
+     * {@link com.jnape.palatable.shoki.api.Natural.Zero} to the {@link Semigroup}.
+     *
+     * @param semigroup the {@link Semigroup}
+     * @param other     the other {@link MultiSet}
+     * @return the merged {@link MultiSet}
+     */
+    default MultiSet<A> merge(Semigroup<Natural> semigroup, MultiSet<A> other) {
+        return foldLeft((acc, a) -> semigroup
+                                .apply(get(a), other.get(a)).match(constantly(acc), result -> acc.add(a, result)),
+                        this.difference(this), // Empty but the correct impl.
+                        unique().union(other.unique()));
+    }
+
+    /**
+     * Common {@link EquivalenceRelation}s between {@link MultiSet MultiSets}.
+     */
+    final class EquivalenceRelations {
+        private EquivalenceRelations() {
+        }
+
+        /**
+         * An {@link EquivalenceRelation} between two {@link MultiSet}s that holds if, and only if, both
+         * {@link MultiSet MultiSets} have the same elements. <code>O(n)</code>.
+         *
+         * @param <A> the element type
+         * @param <S> the {@link MultiSet} subtype of the arguments
+         * @return the {@link EquivalenceRelation}
+         */
+        public static <A, S extends MultiSet<A>> EquivalenceRelation<S> sameElements() {
+            EquivalenceRelation<S> sameMembership = (xs, ys) -> and().foldMap(into(ys::contains), xs);
+            return Sizable.EquivalenceRelations.<S>sameSizes().and(sameMembership);
```

Comment:
Do you want a same sizes check here? If you just care about the unique set, you definitely don't. If you care about the multiplicity, this isn't enough. Which were you thinking?

Reply:
I think it works since it's the overloaded contains but I'm not at an idea right now. It's intended for the multiplicity.

Reply:
Ugh! The stupid dyadic `contains` bit me again. Okay makes sense: let's at least name it `sameElementsSameMultiplicity` though. `sameElements` deceived me.